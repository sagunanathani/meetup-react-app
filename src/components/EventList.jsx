import Event from "./Event";
import PropTypes from "prop-types";

function EventList({ events }) {
  return (
    <div>
      {events.map((event) => (
        <Event key={event.id} event={event} />
      ))}
    </div>
  );
}

EventList.propTypes = {
  events: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      title: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      description: PropTypes.string,
    })
  ).isRequired,
};

export default EventList;
