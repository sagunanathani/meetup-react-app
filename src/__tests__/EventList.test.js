import { render, screen, within } from "@testing-library/react";
import EventList from "../components/EventList";

// ðŸ”¹ Updated mock events to match your current structure
const mockEvents = [
  {
    id: 1,
    summary: "Event 1",
    location: "Berlin, Germany",
    start: { dateTime: "2025-09-10T10:00" },
    created: "2025-09-01T08:00",
  },
  {
    id: 2,
    summary: "Event 2",
    location: "Munich, Germany",
    start: { dateTime: "2025-09-10T12:00" },
    created: "2025-09-01T09:00",
  },
];

describe("<EventList /> component", () => {
  test("renders the list container and correct number of events", () => {
    render(<EventList events={mockEvents} />);

    // ðŸ”¹ Check container exists
    const eventList = screen.getByTestId("event-list");
    expect(eventList).toBeInTheDocument();

    // ðŸ”¹ Check number of list items
    const eventItems = within(eventList).getAllByRole("listitem");
    expect(eventItems).toHaveLength(mockEvents.length);

    // ðŸ”¹ Check content of first event
    expect(eventItems[0]).toHaveTextContent("Event 1");
    expect(eventItems[0]).toHaveTextContent("Berlin, Germany");
    expect(eventItems[0].querySelector("button")).toHaveClass("details-btn");

    // ðŸ”¹ Check content of second event
    expect(eventItems[1]).toHaveTextContent("Event 2");
    expect(eventItems[1]).toHaveTextContent("Munich, Germany");
    expect(eventItems[1].querySelector("button")).toHaveClass("details-btn");
  });

  test("matches updated snapshot", () => {
    const { asFragment } = render(<EventList events={mockEvents} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
