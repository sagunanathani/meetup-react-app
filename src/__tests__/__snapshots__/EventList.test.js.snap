// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<EventList /> component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="events-container"
    data-testid="event-list"
  >
    <li>
      <div
        class="event"
      >
        <h3>
          Event 1
        </h3>
        <p>
          10 Sept 2025, 10:00 am
        </p>
        <p>
          Berlin, Germany
        </p>
        <button>
          Show details
        </button>
      </div>
    </li>
    <li>
      <div
        class="event"
      >
        <h3>
          Event 2
        </h3>
        <p>
          10 Sept 2025, 12:00 pm
        </p>
        <p>
          Munich, Germany
        </p>
        <button>
          Show details
        </button>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
